android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageExpoMigration.java

```diff
@@ -0,0 +1,154 @@
+            return;
+        }
+
+        ArrayList<File> expoDatabases = getExpoDatabases(context);
+
+        File expoDatabase = getLastModifiedFile(expoDatabases);
+
+        if (expoDatabase == null) {
+            Log.v(LOG_TAG, "No scoped database found");
+            return;
+        }
+
+        try {
+            // Create the storage file
+            ReactDatabaseSupplier.getInstance(context).get();
+            copyFile(new FileInputStream(expoDatabase), new FileOutputStream(context.getDatabasePath(ReactDatabaseSupplier.DATABASE_NAME)));
+            Log.v(LOG_TAG, "Migrated most recently modified database " + expoDatabase.getName() + " to RKStorage");
+        } catch (Exception e) {
+            Log.v(LOG_TAG, "Failed to migrate scoped database " + expoDatabase.getName());
+            e.printStackTrace();
+            return;
+        }
+
+        try {
+            for (File file : expoDatabases) {
+                if (file.delete()) {
+                    Log.v(LOG_TAG, "Deleted scoped database " + file.getName());
+                } else {
+                    Log.v(LOG_TAG, "Failed to delete scoped database " + file.getName());
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        Log.v(LOG_TAG, "Completed the scoped AsyncStorage migration");
+    }
+
+    private static boolean isAsyncStorageDatabaseCreated(Context context) {
+        return context.getDatabasePath(ReactDatabaseSupplier.DATABASE_NAME).exists();
+    }
+
+    // Find all database files that the user may have created while using Expo.
+    private static ArrayList<File> getExpoDatabases(Context context) {
+        ArrayList<File> scopedDatabases = new ArrayList<>();
+        try {
+            File databaseDirectory = context.getDatabasePath("noop").getParentFile();
+            File[] directoryListing = databaseDirectory.listFiles();
+            if (directoryListing != null) {
+                for (File child : directoryListing) {
+                    // Find all databases matching the Expo scoped key, and skip any database journals.
+                    if (child.getName().startsWith("RKStorage-scoped-experience-") && !child.getName().endsWith("-journal")) {
+                        scopedDatabases.add(child);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            // Just in case anything happens catch and print, file system rules can tend to be different across vendors.
+            e.printStackTrace();
+        }
+        return scopedDatabases;
+    }
+
+    // Returns the most recently modified file.
+    // If a user publishes an app with Expo, then changes the slug 
+    // and publishes again, a new database will be created. 
+    // We want to select the most recent database and migrate it to RKStorage.
+    private static File getLastModifiedFile(ArrayList<File> files) {
+        if (files.size() == 0) {
```

Comment:
Would it make sense to add a special case for `files.size() == 1`?

Reply:
doesn't seem necessary, even the case for zero isn't strictly required.